op-node/rollup/derive/attributes.go

```diff
@@ -102,28 +102,35 @@ func (ba *FetchingAttributesBuilder) PreparePayloadAttributes(ctx context.Contex
 			l2Parent, nextL2Time, eth.ToBlockID(l1Info), l1Info.Time()))
 	}
 
-	l1InfoTx, err := L1InfoDepositBytes(ba.rollupCfg, sysConfig, seqNumber, l1Info, nextL2Time)
-	if err != nil {
-		return nil, NewCriticalError(fmt.Errorf("failed to create l1InfoTx: %w", err))
+	var upgradeTxs []hexutil.Bytes
+	if ba.rollupCfg.IsEcotoneActivationBlock(nextL2Time) {
+		upgradeTxs, err = EcotoneNetworkUpgradeTransactions()
+		if err != nil {
+			return nil, NewCriticalError(fmt.Errorf("failed to build ecotone network upgrade txs: %w", err))
+		}
 	}
 
-	// If this is the Ecotone activation block we update the system config by copying over "Scalar"
-	// to "BasefeeScalar". Note that after doing so, the L2 view of the system config differs from
-	// that on the L1 up until we receive a "type 4" log event that explicitly updates the new
-	// scalars.
-	if ba.rollupCfg.IsEcotoneActivationBlock(nextL2Time) {
-		// check if the scalar is too big to convert to uint32, and if so just use the uint32 max value
-		basefeeScalar := uint32(math.MaxUint32)
+	// If we are past the activation block, then we use the new Ecotone L1-info data,
+	// and need to migrate our pre-ecotone scalar into the new ecotone scalar.
+	if ba.rollupCfg.IsEcotoneActivationBlock(l2Parent.Time) {
```

Comment:
Why do the scalar migration only after the activation block? Seems safer to do it in the same clause above when you hit the activation block?  (Could an ecotone gas_config update event come in at the same time as the activation that would then get overwritten by this delay in migration?)

Reply:
this changed with the new more compatible scalar migration